artifactor/plugins/reporter.py

```diff
@@ -248,7 +249,12 @@ def build_dict(self, path, container, contents):
         Build a hierarchical dictionary including information about the stats at each level
         and the duration.
         """
-        segs = path.split('/')
+
+        if isinstance(path, basestring):
+            segs = process_pytest_path(path)
+        else:
+            segs = path
```

Comment:
Well, the build_dict calls itself recursively, so when it once breaks the string to the path, we don't need to convert it in the subsequent calls, we just pass the resst of the list.